Formula/ghc@9.rb

```diff
@@ -22,6 +22,13 @@ class GhcAT9 < Formula
   depends_on "sphinx-doc" => :build
 
   # https://www.haskell.org/ghc/download_ghc_9_0_2.html#macosx_x86_64
+  uses_from_macos "m4" => :build
+  uses_from_macos "ncurses"
+
+  on_linux do
+    depends_on "gmp" => :build
```

Comment:
This doesn't seem like the right fix. GHC 9 is meant to have an integer backend that doesn't need GMP anymore.

Reply:
You're right that the bootstrap binary does not require GMP to run any more, so that part is not needed.  The situation with GMP for the copy of GHC we build seems a bit more complicated.

When I tried to remove `depends_on "gmp"` altogether, the build of `ghc-cabal` fails because it still tries to link `-lgmp`.  This happens even if I specify `--without-gmp` to `configure`.  It seems `--with-intree-gmp` is still needed as well, otherwise it will dynamically link to `gmp` and propagate a runtime dependency on `gmp` to all Haskell formulae that we have worked to avoid.

According to https://iohk.io/en/blog/posts/2020/07/28/improving-haskells-big-numbers-support/, it seems that while `native-integer` is a major improvement of `simple-integer` as a native Haskell integer backend, is still inferior to `integer-gmp` in performance.  So it may not be a good idea to stop using GMP altogether even if we can figure out how to do it.